internal/pkg/aws/ecs/ecs.go

```diff
@@ -1,36 +1,75 @@
-// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
 // Package ecs provides a client to make API requests to Amazon Elastic Container Service.
 package ecs
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ecs"
 )
 
+const (
+	shortTaskIDLength      = 7
+	shortImageDigestLength = 7
+	imageDigestPrefix      = "sha256:"
+)
+
 type ecsClient interface {
+	DescribeTasks(input *ecs.DescribeTasksInput) (*ecs.DescribeTasksOutput, error)
 	DescribeTaskDefinition(input *ecs.DescribeTaskDefinitionInput) (*ecs.DescribeTaskDefinitionOutput, error)
+	DescribeServices(input *ecs.DescribeServicesInput) (*ecs.DescribeServicesOutput, error)
+	ListTasks(input *ecs.ListTasksInput) (*ecs.ListTasksOutput, error)
 }
 
-// Service wraps an AWS ECS client.
-type Service struct {
-	ecs ecsClient
+// ECS wraps an AWS ECS client.
+type ECS struct {
+	client ecsClient
+}
+
+// TaskDefinition wraps up ECS TaskDefinition struct.
+type TaskDefinition ecs.TaskDefinition
+
+// Service wraps up ECS Service struct.
+type Service ecs.Service
+
+// Task wraps up ECS Task struct.
+type Task ecs.Task
+
+// ServiceStatus contains the status info of a service.
+type ServiceStatus struct {
+	DesiredCount int64
+	RunningCount int64
+	Status       string
+}
+
+// TaskStatus contains the status info of a task.
+type TaskStatus struct {
+	DesiredStatus string
+	ID            string
+	Images        []string
+	ImageDigests  []string
```

Comment:
what do you think of:
```suggestion
	Images        []struct {
		ID string
		Digest string
	}
```
or having an `Image` struct?

Reply:
Good idea!